crates/solver/src/settlement.rs

```diff
@@ -299,6 +318,77 @@ impl Settlement {
         }
     }
 
+    // Checks whether the settlement prices do not deviate more than max_settlement_price_deviation from the auction prices on certain pairs.
+    pub fn satisfies_price_checks(
+        &self,
+        auction_id: u64,
+        solver_name: &str,
+        external_prices: &ExternalPrices,
+        max_settlement_price_deviation: &Ratio<BigInt>,
+        tokens_to_satisfy_price_test: &TokensForPriceCheck,
+    ) -> bool {
+        if matches!(tokens_to_satisfy_price_test, TokensForPriceCheck::Tokens(token_list) if token_list.is_empty())
```

Comment:
Maybe I'm a bit rusty but I can not tell what this is trying to do. Maybe it has something to do with the long names... or the if statement on the inside without an else... I don't know

Reply:
`matches` syntax is just sugar for:

```rust
if match tokens_to_satisfy_price_test {
    TokensForPriceCheck::Tokens(token_list) if token_list.is_empty() => true,
    _ => false,
}  {
    // ...
}
```

I think `clippy` even suggests this construct in some cases.